### PR TITLE
fix(nuxt): add `baseURL` to island fetch requests

### LINKED ISSUE #22002: Server components break if you set baseURL
### Environment

- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  npm@9.4.2
- Builder:          vite
- User Config:      app, experimental
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://github.com/Twitch0125/nuxt-server-components-base-url-reproduction

I'd add a stackblitz link, but I think stackblitz's github integration might be broken at the time of creating this?

### Describe the bug

when defining `app.baseURL` in nuxt.config.ts, server components don

I also see this warning server-side
```
 [Vue warn]: Component NuxtIsland is missing template or render function.

### Additional context

_No response_

### Logs

_No response_

### LINKED ISSUE #22010: Prerendering is not working with `app.baseURL`
### Environment

------------------------------
- Operating System: Windows_NT
- Node Version:     v18.16.1
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  pnpm@7.9.0
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

stackblitz reproduction incoming

-> fixtures
set a baseURL for the `fixtures/basic` in the nuxt repo 
run ` pnpm nuxi build .\test\fixtures\basic\`

### Describe the bug

We got logs indicating that prerender is failing

````
✔ Server built in 5809ms                                                                                                                                                                         10:33:49
✔ Generated public test/fixtures/basic/.output/public                                                                                                                                      nitro 10:33:50
ℹ Initializing prerenderer                                                                                                                                                                 nitro 10:33:50
ℹ Prerendering 3 routes                                                                                                                                                                    nitro 10:34:04
  ├─ /random/a (1699ms) (Error: [500] )                                                                                                                                                     nitro 10:34:06
  ├─ /random/b (623ms) (Error: [500] )                                                                                                                                                      nitro 10:34:06
                                                                                                                                                                                            nitro 10:34:06  
Errors prerendering:
  ├─ /random/a (500)                                                                                                                                                                        nitro 10:34:06  
  ├─ /random/b (500)                                                                                                                                                                        nitro 10:34:06  
  ├─ /random/c (500)                                                                                                                                                                        nitro 10:34:06  
                       
````

EDIT: building `test/fixtures/basic` crash due to this issue 

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
resolve https://github.com/nuxt/nuxt/issues/22002
resolve #22010

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Hi :wave: this PR prepend the fetch request of `NuxtIsland` with the baseURL or it wouldn't correctly hit the endpoint of islands renderer

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

@danielroe can i set a `app.baseURL` for `fixtures/basic` ?
- edit: building it with `app.baseURL` crash due to prerender -> https://github.com/nuxt/nuxt/issues/22010

### todo 
- [x] non-reg test with `NUXT_APP_BASE_URL` within fixtures tests

### CODE REVIEW

File: packages/nuxt/src/app/components/nuxt-island.ts [hunk 1]
```diff
@@ -79,7 +81,7 @@ export default defineComponent({
         appendResponseHeader(event, 'x-nitro-prerender', url)
       }
       // TODO: Validate response
-      const r = await eventFetch(withQuery(url, {
+      const r = await eventFetch(withQuery(join(appConfig.app.baseURL ?? '', url), {
```
[the remark this answers is not in the document]
Comment (huang-julien) [reply packages/nuxt/src/app/components/nuxt-island.ts:84]: hmmm we can't really build the `test/fixtures/basic` directory due to https://github.com/nuxt/nuxt/issues/22010. But the issue is also happening in dev

[the remark this answers is not in the document]
Comment (danielroe) [reply packages/nuxt/src/app/components/nuxt-island.ts:84]: Ah, missed that.

We'll have to do this the other way then - _omit_ the baseURL for dev+client.

**Comment by danielroe** [comment]
Regarding setting `app.baseURL` we do this already dynamically in one of the tests - maybe you could reproduce the issue in a regression test that way?

**Comment by huang-julien** [comment]
Sure ! I'll convert this PR to draft. Will finish in a few days

**Review by danielroe — APPROVED** [review]

**Pull request merged**